bigquery/src/main/java/org/apache/zeppelin/bigquery/bigQueryInterpreter.java

```diff
@@ -0,0 +1,322 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.bigquery;
+
+import static org.apache.commons.lang.StringUtils.containsIgnoreCase;
+
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.json.jackson2.JacksonFactory;
+
+import com.google.api.services.bigquery.Bigquery;
+import com.google.api.services.bigquery.BigqueryScopes;
+import com.google.api.client.json.GenericJson;
+import com.google.api.services.bigquery.Bigquery.Datasets;
+import com.google.api.services.bigquery.BigqueryRequest;
+import com.google.api.services.bigquery.model.DatasetList;
+import com.google.api.services.bigquery.model.Job;
+import com.google.api.services.bigquery.model.TableCell;
+import com.google.api.services.bigquery.model.TableFieldSchema;
+import com.google.api.services.bigquery.model.TableRow;
+import com.google.api.services.bigquery.model.TableSchema;
+import com.google.api.services.bigquery.Bigquery.Jobs.GetQueryResults;
+import com.google.api.services.bigquery.model.GetQueryResultsResponse;
+import com.google.api.services.bigquery.model.QueryRequest;
+import com.google.api.services.bigquery.model.QueryResponse;
+import com.google.gson.Gson;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.util.Properties;
+import java.util.Set;
+
+import org.apache.zeppelin.interpreter.Interpreter;
+import org.apache.zeppelin.interpreter.InterpreterContext;
+import org.apache.zeppelin.interpreter.InterpreterPropertyBuilder;
+import org.apache.zeppelin.interpreter.InterpreterResult;
+import org.apache.zeppelin.interpreter.InterpreterResult.Code;
+import org.apache.zeppelin.interpreter.thrift.InterpreterCompletion;
+import org.apache.zeppelin.scheduler.Scheduler;
+import org.apache.zeppelin.scheduler.SchedulerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.google.common.collect.Sets.SetView;
+import java.io.PrintStream;
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+/**
+ * BigQuery interpreter for Zeppelin.
+ * 
+ * <ul>
+ * <li>{@code bigquery.project_id} - Project ID in GCP</li>
+ * </ul>
+ * 
+ * <p>
+ * How to use: <br/>
+ * {@code %bqsql.sql<br/>
+ * {@code
+ *  SELECT departure_airport,count(case when departure_delay>0 then 1 else 0 end) as no_of_delays 
+ *  FROM [bigquery-samples:airline_ontime_data.flights] 
+ *  group by departure_airport 
+ *  order by 2 desc 
+ *  limit 10
+ * }
+ * </p>
+ * 
+ */
+
```

Comment:
The above comments are better for `bigquery.md` as @bzz  said in [https://github.com/apache/zeppelin/pull/1170#issuecomment-232215164](https://github.com/apache/zeppelin/pull/1170#issuecomment-232215164). You can refer to many existed docs [in here](https://github.com/apache/zeppelin/tree/master/docs/interpreter). It would be helpful to users who want to try BigQuery in Zeppelin.